lib/taskana-core/src/main/java/pro/taskana/task/internal/ServiceLevelHandler.java

```diff
@@ -325,11 +340,12 @@ private Instant getFollowingWorkingDays(Instant instant, Duration days) {
    */
   private void ensureServiceLevelIsNotViolated(
       TaskImpl task, Duration duration, Instant calcPlanned) throws InvalidArgumentException {
+    // TODO tests mit coverage. falls die Exception nie auftritt weg mit der Methode
     if (task.getPlanned() != null
         && !task.getPlanned().equals(calcPlanned)
         // manual entered planned date is a different working day than computed value
-        && (converter.isWorkingDay(task.getPlanned())
-            || converter.hasWorkingDaysInBetween(task.getPlanned(), calcPlanned))) {
+        && (workingTimeCalculator.isWorkingDay(task.getPlanned())
+            || workingTimeCalculator.isWorkingTimeBetween(task.getPlanned(), calcPlanned))) {
```

Comment:
@mustaphazorgati  If you have any idea how we can make this code nice *and* use a version of `isWorkingTimeBetween` that expects the first parameter to be before the second one, let me know. All I can think of is quite clumsy.

Reply:
We decided to keep it as it is